### PR TITLE
[PyTorch] Aten Layer Norm parser & Layer Norm LibNode

### PR DESCRIPTION
**Pull request by Atrisan**
## Description
Adds a new Library node to represent a layer norm.
Adds support for the aten.native_layer_norm.default operation with a new parser.

## Related Issues

- Closes #

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor / internal change (no functional change)
- [ ] Documentation update

## Quality Checklist
*Please ensure the following are completed before requesting review:*

* [x] Runtime-compatible (no externally visible existing method has its parameters changed, was deleted or has its exception behavior changed)
* [x] Compile-time compatible (new arguments & properties added have default values, but users need to recompile against the new version to work)

IR:
* [ ] Introduces new types into the SDFG IR
* [x]  Changed / added properties on SDFG IR Elements (LibraryNodes, ControlFlowNodes DataFlowNodes etc.)
  *  [x] New / modified types have serialization / deserialization support
  *  [x] Deserialization is backward compatible (new serialization can deserialize old format and represent it in the new format if applicable)
  *  [x] Serialized format is backward compatible (previous deserializer can read new serialized format without incorrectness (just lacking additional guarantees that are optional or did not exist before)
  *  [x] Serializer has an option to emit the previous format

### Unit Tests
- [x] New unit tests have been added for the changes.
- [x] Existing unit tests pass locally.
- [x] Edge cases and error paths are covered.

### Integration Tests
- [x] New integration tests have been added (or existing ones updated).
- [x] Integration tests pass locally.
- [x] Interactions with other components/services are verified.

### Documentation
- [ ] Public APIs, options, and behavior changes are documented.
- [x] README / docs site / inline comments updated where applicable.
- [ ] Changelog / release notes updated (if applicable).

### Test Coverage
- [x] Test coverage has not decreased as a result of this change.
- [x] Coverage report reviewed and acceptable.
- [x] Critical / new code paths are covered.

## How Has This Been Tested?
Describe the tests you ran to verify your changes and provide instructions to reproduce.
Include relevant details for your test configuration (OS, Python/LLVM/Docc version, hardware/target).

## Additional Context
Add any other context about the PR here (design decisions, trade-offs, follow-ups).

### CODE REVIEW

File: pytorch/docc/pytorch/graph_parser/normalization.py [hunk 1]
```diff
@@ -97,3 +97,81 @@ def parse(
...
+        node: torch.fx.Node,
+        builder: StructuredSDFGBuilder,
+        container_info: ContainerInfos,
+    ) -> None:
+        if len(node.args) < 5:
+            raise GraphParserError(
+                self,
+                node,
+                "Expected at least 5 arguments but got " + str(len(node.args)),
+            )
```
Comment (Moehre2) [inline pytorch/docc/pytorch/graph_parser/normalization.py:114]: The Core ATen IR op requires _exactly_ 5 arguments

**Review by Moehre2 — DISMISSED** [review]
Passing empty strings to a method called `add_layernorm_with_bias` to not add bias is a little bit dirty but OK for now

**Review by Moehre2 — APPROVED** [review]
